### PR TITLE
Feature/better saturation system

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #

# Changes proposed in this pull request:
- Improved saturation system, no more daily reductions, but reductions for meters walked and for actions done.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 1]
```diff
@@ -234,14 +289,18 @@ private void initTasks()
         this.tasks.addTask(++priority, new EntityAISleep(this));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:289]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 2]
```diff
@@ -234,14 +289,18 @@ private void initTasks()
...
         this.tasks.addTask(++priority, new EntityAIOpenDoor(this, true));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:290]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 3]
```diff
@@ -73,28 +73,16 @@
...
-import static com.minecolonies.api.util.constant.Suppression.UNCHECKED;
-import static com.minecolonies.api.util.constant.TranslationConstants.CITIZEN_RENAME_NOT_ALLOWED;
-import static com.minecolonies.api.util.constant.TranslationConstants.CITIZEN_RENAME_SAME;
+import static com.minecolonies.api.util.constant.Suppression.*;
+import static com.minecolonies.api.util.constant.TranslationConstants.*;
 
 /**
  * The Class used to represent the citizen entities.
  */
 public class EntityCitizen extends EntityAgeable implements INpc
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:84]: MAJOR Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 67 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
MAJOR class "EntityCitizen" has 105 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 4]
```diff
@@ -234,14 +289,18 @@ private void initTasks()
         this.tasks.addTask(++priority, new EntityAISleep(this));
         this.tasks.addTask(++priority, new EntityAIOpenDoor(this, true));
         this.tasks.addTask(priority, new EntityAIOpenFenceGate(this, true));
-        this.tasks.addTask(++priority, new EntityAIWatchClosest2(this, EntityPlayer.class, 3.0F, 1.0F));
-        this.tasks.addTask(++priority, new EntityAIWatchClosest2(this, EntityCitizen.class, 5.0F, 0.02F));
-        this.tasks.addTask(++priority, new EntityAICitizenWander(this, 0.6D));
-        this.tasks.addTask(++priority, new EntityAIWatchClosest(this, EntityLiving.class, 6.0F));
+        this.tasks.addTask(++priority, new EntityAIWatchClosest2(this, EntityPlayer.class, WATCH_CLOSEST2, 1.0F));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:292]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 5]
```diff
@@ -234,14 +289,18 @@ private void initTasks()
...
+        this.tasks.addTask(++priority, new EntityAIWatchClosest2(this, EntityCitizen.class, WATCH_CLOSEST2_FAR, WATCH_CLOSEST2_FAR_CHANCE));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:293]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 6]
```diff
@@ -234,14 +289,18 @@ private void initTasks()
...
+        this.tasks.addTask(++priority, new EntityAICitizenWander(this, DEFAULT_SPEED));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:294]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)

File: src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java [hunk 7]
```diff
@@ -234,14 +289,18 @@ private void initTasks()
...
+        this.tasks.addTask(++priority, new EntityAIWatchClosest(this, EntityLiving.class, WATCH_CLOSEST));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:295]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS881)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 90 issues
* ![MAJOR][MAJOR] 82 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/14b7bc549e5f82ffc400f4f2534d473eb8400e02/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**